### ISSUE TITLE
Apparently random undefined statusCode error

### ISSUE THREAD

**Issue by mircolac** [Marketcloud/marketcloud-node#1]
Sometimes, apparently random, the SDK gives me this error, ending in a node server crash.   

```
 if (401 === err.response.statusCode){
                                        ^

TypeError: Cannot read property 'statusCode' of undefined
    at C:\ProjectFolder\node_modules\marketcloud-node\index.js:131:30
at Request.callback (C:\ProjectFolder\node_modules\superagent\lib\node\index.js:687:12)
at ClientRequest.<anonymous> (C:\ProjectFolder\node_modules\superagent\lib\node\index.js:639:10)
at emitOne (events.js:96:13)
at ClientRequest.emit (events.js:188:7)
at TLSSocket.socketErrorListener (_http_client.js:308:9)
at emitOne (events.js:96:13)
at TLSSocket.emit (events.js:188:7)
at emitErrorNT (net.js:1271:8)
at _combinedTickCallback (internal/process/next_tick.js:74:11)
at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by mircolac** [comment]
The error is a ETIMEDOUT exception!

**Comment by fatmatto** [comment]
The error is due to a wrong handling of the error response, we should check for a response inside the error object before actually checking the statusCode.
Fixed in 02e0bc04261c5f31d1bb9cc7c7ee31825a89f2cb
